src/os/bluestore/KernelDevice.cc

```diff
@@ -659,6 +748,14 @@ int KernelDevice::aio_write(
   return 0;
 }
 
+int KernelDevice::discard(uint64_t offset, uint64_t len)
+{
+  int r = 0;
+  if (!rotational)
+    r = block_device_discard(fd_direct, (int64_t)offset, (int64_t)len);
+  return r;
+}
```

Comment:
@wido @k0ste 
I had already implemented that part.
You all don't have to worry about issuing discard command to rotational device.